disk-buffering/src/test/java/io/opentelemetry/contrib/disk/buffering/exporters/MetricDiskExporterTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.exporters;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import io.opentelemetry.contrib.disk.buffering.internal.serialization.serializers.SignalSerializer;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.TestData;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class MetricDiskExporterTest {
+
+  private MetricExporter wrapped;
+  private MetricDiskExporter exporter;
+  @TempDir File rootDir;
+
+  @BeforeEach
+  public void setUp() {
+    wrapped = mock();
+    exporter = new MetricDiskExporter(wrapped, rootDir, TestData.getDefaultConfiguration());
+  }
+
+  @Test
+  public void verifyStorageFolderName() {
+    assertEquals("metrics", exporter.getStorageFolderName());
+  }
+
+  @Test
+  public void callWrappedWhenDoingExport() {
+    List<MetricData> data = Collections.emptyList();
+    CompletableResultCode result = CompletableResultCode.ofSuccess();
+    doReturn(result).when(wrapped).export(data);
+
+    assertEquals(result, exporter.doExport(data));
+
+    verify(wrapped).export(data);
+  }
+
+  @Test
+  public void verifySerializer() {
+    assertEquals(SignalSerializer.ofMetrics(), exporter.getSerializer());
+  }
+
+  @Test
+  public void onFlush_flushWrappedExporter() {
+    exporter.flush();
+
+    verify(wrapped).flush();
+  }
+
+  @Test
+  public void provideWrappedAggregationTemporality() {
```

Comment:
a little bit too fine granular test - but it's a matter of preference
In other words, tests that only check that a particular method is forwarded are not very valuable IMO.

Reply:
Trust me I want to spend time writing more tests than needed 😄 so I understand your point, though since the main focus of this implementation is to be a wrapper, I wanted to make it bulletproof in case somebody might remove this method call forwarding by mistake in the future (even myself, I tend to forget about the details of my code after like 2 months 😅)